### PR TITLE
Add more HTTP status codes

### LINKED ISSUE #12591: Issue with "HTTP response status codes": some Http status codes not documented
MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

#### What information was incorrect, unhelpful, or incomplete?

All below mentioned http status code is missing from the left navigation pane in the references section of HTTP response status codes.

102 Processing (WebDAV), 207 Multi-Status (WebDAV),208 Already Reported (WebDAV),226 IM Used (HTTP Delta encoding),421 Misdirected Request,423 Locked (WebDAV),424 Failed Dependency (WebDAV)

#### Specific section or headline?
references section of HTTP response status codes.

#### What did you expect to see?

All the said response codes to be listed.

#### Did you test this? If so, how?

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/http/status`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/http/status/index.md
* Last commit: https://github.com/mdn/content/commit/cdfd4547c1d5ec29d87229ce187a4a401289f335
* Document last modified: 2021-11-18T03:30:28.000Z

</details>

### PR DESCRIPTION
**Pull request by Harshil-Jani**
#### Summary

HTTP response status codes.
- 102 Processing 
- 207 Multi-Status
- 208 Already Reported
- 226 IM Used 
- 421 Misdirected Request
- 423 Locked
- 424 Failed Dependency 
were missing in the list. So they were added.
#### Motivation

This will be my very first contribution towards MDN. I used it as a major resource of learning so now with this I want to be part of the MDN network and help it grow more effectively
#### Related issues

Fixes #12591 

#### Metadata

- [x] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

**Comment by Muradalsaaideh** [comment]
> 
> #### Summary
> 
> HTTP response status codes.
> - 102 Processing 
> - 207 Multi-Status
> - 208 Already Reported
> - 226 IM Used 
> - 421 Misdirected Request
> - 423 Locked
> - 424 Failed Dependency 
> were missing in the list. So they were added.
> #### Motivation
> 
> This will be my very first contribution towards MDN. I used it as a major resource of learning so now with this I want to be part of the MDN network and help it grow more effectively
> #### Supporting details
> 
> 
> #### Related issues
> 
> Fixes #12591 
> 
> #### Metadata
> 
> - [x] Adds a new document
> - [ ] Rewrites (or significantly expands) a document
> - [ ] Fixes a typo, bug, or other error
> 
> 
> 

e889efc4e3a59db1525f4001ac6be3189bef5968

**Comment by Muradalsaaideh** [comment]
> > 
> > #### Summary
> > 
> > HTTP response status codes.
> > - 102 Processing 
> > - 207 Multi-Status
> > - 208 Already Reported
> > - 226 IM Used 
> > - 421 Misdirected Request
> > - 423 Locked
> > - 424 Failed Dependency 
> > were missing in the list. So they were added.
> > #### Motivation
> > 
> > This will be my very first contribution towards MDN. I used it as a major resource of learning so now with this I want to be part of the MDN network and help it grow more effectively
> > #### Supporting details
> > 
> > 
> > #### Related issues
> > 
> > Fixes #12591 
> > 
> > #### Metadata
> > 
> > - [x] Adds a new document
> > - [ ] Rewrites (or significantly expands) a document
> > - [ ] Fixes a typo, bug, or other error
> > 
> > 
> > 
> 
> e889efc4e3a59db1525f4001ac6be3189bef5968

e889efc4e3a59db1525f4001ac6be3189bef5968

**Comment by teoli2003** [comment]
Before reviewing in detail, @hamishwillee and @elchi3, what is your opinion about adding these as they are part of the (fairly common) WebDAV specs and not the web _stricto sensu_.

I'm torn here.

**Comment by bhaveshraj336** [comment]
@teoli2003 I can make any changes needed ?
The PR is been on hold from last 6 months?

**Comment by teoli2003** [comment]
We are waiting for answers from @hamishwillee and @elchi3 here.

**Comment by estelle** [comment]
pinging @hamishwillee and @Elchi3

File: files/en-us/web/http/status/208/index.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+## Status
+
+```
+208 Already Reported
+```
+
+## Specifications
+
+| Specification                         | Status   | Comments           |
```
Comment (hamishwillee) [inline files/en-us/web/http/status/208/index.md:23]: Use spec URLs

**Comment by hamishwillee** [comment]
@teoli2003 @estelle Sorry to be late to the party. FWIW I agree with @Elchi3 

Specifically, I use MDN for first place to look up this kind of information and I think this is a good contribution (thanks @Harshil-Jani). 

But I'd like not to have those spec tables in the source. Is it possible to specify a "spec_url" in frontmatter yet? I couldn't find an example of this in content.

I'm also not sure about omitting the compatibility section. My concern here is that readers don't necessarily understand that it is missing because there is no compatibility story to tell. Not, not so worried about this case, but see case in aside below.

@teoli2003 I assume you are going to review this following the feedback that we think this is good, but if you want me to then let me know.

---

Bigger picture aside ...

W.r.t. browser compatibility we have the same problem with some HTTP headers in that "browser support" is nonsensical. How we indicate this seems to be inconsistent - there are plenty of cases where we just say "true". It would be good to agree an approach and use it here.

The obvious thought is that if something has no compatibility story then it shouldn't be in BCD, and that we therefore should push the spec urls into page metadata. I dislike this because there is no way to know if something is missing or deliberately omitted from BCD. Also BCD is good at managing the spec data and having it managed in multiple places sux.

So if it were practical and politically possible, my preference would be to work out a way to indicate "not applicable" for version compatibility and still keep the specs in BCD.

**Comment by hamishwillee** [comment]
@teoli2003 I put a comment above, but I think this is still waiting on you.

**Comment by hamishwillee** [comment]
OK, I'm going to ping @teoli2003 one last time. If not, next time I come here I'm going to do the work to get this in.

File: files/en-us/web/http/status/208/index.md (hunk 1 above)
Comment (teoli2003) [reply files/en-us/web/http/status/208/index.md:23]:
```suggestion
{{Specifications}}
```

File: files/en-us/web/http/status/208/index.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+---
+title: 208 Already Reported
+slug: Web/HTTP/Status/208
+tags:
+  - HTTP
+  - Status code
+  - Reported
+browser-compat: http.status.208
```
Comment (teoli2003) [inline files/en-us/web/http/status/208/index.md:8]:
```suggestion
browser-compat: http.status.208
spec-url: https://www.rfc-editor.org/rfc/rfc5842.html#section-7.1
```

**Comment by teoli2003** [comment]
Ok, I got a look at this, and I'll add a few commits to get specs and compatibility in. As these are not in bcd, there is no compatibility per se, but I'll put a sentence explaining why.

Then I'll hand it to Hamish for a last review before landing this.

**Review by teoli2003 — APPROVED** [review]
This is now ok for me. Let's get a last review from Hamish then land this.

File: files/en-us/web/http/status/207/index.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+tags:
+  - HTTP
+  - Status code
+  - Partial Success
+spec-urls: https://www.rfc-editor.org/rfc/rfc4918#section-11.1
+---
+
+{{HTTPSidebar}}
+
+The HTTP **`207 Multi Status`** response code indicates that there might be mixture of responses. We use {{HTTPStatus(204)}} if all the operations succededs or else if all fails we can use {{HTTPStatus(403)}}. But if half of them fails and half succedds we use Multi Status in such conditions.
```
Comment (hamishwillee) [inline files/en-us/web/http/status/207/index.md:13]: @teoli2003 I fixed the typo and English  here, but I'm not sure what "multi status" means below.

```suggestion
The HTTP **`207 Multi Status`** response code indicates that there might be mixture of responses.
We use {{HTTPStatus(204)}} if all the operations succeed and {{HTTPStatus(403)}} if all operations fail.
If some operations succeed and others fail then we use Multi Status.
```

File: files/en-us/web/http/status/207/index.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+---
+
+{{HTTPSidebar}}
+
+The HTTP **`207 Multi Status`** response code indicates that there might be mixture of responses.
+
+We use {{HTTPStatus(204)}} if all the operations succeed and {{HTTPStatus(403)}} if all operations fail.
+`207 Multi Status` is returned if some operations succeed and others fail.
+
+If you perform a destructive operation like {{HTTPMethod("POST")}}, {{HTTPMethod("DELETE")}} etc. against more than one resource and the operations against each individual resource does not share a common outcome then you can go for 207.
```
Comment (hamishwillee) [inline files/en-us/web/http/status/207/index.md:18]: I thought each operation was atomic so if you deleted a resource it would succeed or fail in entirety. I assume this is not the case. Is this wording correct?

```suggestion
If you perform a destructive operation like {{HTTPMethod("POST")}}, {{HTTPMethod("DELETE")}} etc. against more than one resource and the operations against each individual resource do not not share a common outcome then the response can be `207`.
```

File: files/en-us/web/http/status/208/index.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+tags:
+  - HTTP
+  - Status code
+  - Reported
+spec-urls: https://www.rfc-editor.org/rfc/rfc5842.html#section-7.1
+---
+
+{{HTTPSidebar}}
+
+The HTTP **`208 Already Reported`** response code is used inside DAV: propstat response element. This uses Bindings, so to avoid mapping of internal members of multiple binding to a collection inside the request's scope, only one will be reported with {{HTTPStatus(200)}} while other responses for all other bindings will use {{HTTPStatus(208)}} so no conflicts are created.
```
Comment (hamishwillee) [inline files/en-us/web/http/status/208/index.md:13]: @teoli2003 Is this sufficient? As a lay reader I do not know anything about DAV, Bindings, internal members etc. So this is "spec language" to me.

File: files/en-us/web/http/status/102/index.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
+---
+title: 102 Processing
+slug: Web/HTTP/Status/102
```
Comment (hamishwillee) [inline files/en-us/web/http/status/102/index.md:3]: @wbamberg While we're here, is there an agreed set of HTTP page types yet?

File: files/en-us/web/http/status/226/index.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+tags:
+  - HTTP
+  - Status code
+  - Used
+spec-urls: https://www.rfc-editor.org/rfc/rfc3229.html#section-10.4.1
+---
+
+{{HTTPSidebar}}
+
+The HTTP **`226 IM Used`** response code is used when the server has fulfilled a {{HTTPMethod("GET")}} Request for the resource but the response is a representation of the result of one or more instance manipulations applied by combining with previous or future responses as specific instances.
```
Comment (hamishwillee) [inline files/en-us/web/http/status/226/index.md:13]:
```suggestion
The HTTP **`226 IM Used`** response code is used when the server has fulfilled a {{HTTPMethod("GET")}} request for the resource but the response is a representation of the result of one or more instance manipulations applied by combining with previous or future responses as specific instances.
```

Comment (hamishwillee) [inline files/en-us/web/http/status/226/index.md:13]: I don't know what this means. Is it sufficient?

File: files/en-us/web/http/status/421/index.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+tags:
+  - HTTP
+  - HTTP Status Code
+  - Reference
+spec-urls: https://www.rfc-editor.org/rfc/rfc9110#name-421-misdirected-request
+---
+
+{{HTTPSidebar}}
+
+The HTTP **`421 Misdirected Request`** client error response code indicates that the request was directed at a server that is not able to produce a response. This might be possible if a connection is reused or if an alternative service is selected.
```
Comment (hamishwillee) [inline files/en-us/web/http/status/421/index.md:13]:
```suggestion
The HTTP **`421 Misdirected Request`** client error response code indicates that the request was directed to a server that is not able to produce a response. This might be possible if a connection is reused or if an alternative service is selected.
```

Comment (hamishwillee) [inline files/en-us/web/http/status/421/index.md:13]: @teoli2003 What is a client error response code? I.e. I know that a response comes from a server in response to a request - so all responses are from servers? Or is this somehow internally generated?

Basically I am not sure who provides this response, or how it might happen.

File: files/en-us/web/http/status/423/index.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  - HTTP
+  - HTTP Status Code
+  - Reference
+  - Client Error
+spec-urls: https://www.rfc-editor.org/rfc/rfc4918#section-11.3
+---
+
+{{HTTPSidebar}}
+
+The HTTP **`423 Locked`** client error response code indicates that either the source or the destination of a method is locked. It should contain some conditions such that the accessible resources are not locked in order to access them or to prevent any modifications they must be locked.
```
Comment (hamishwillee) [inline files/en-us/web/http/status/423/index.md:14]: I understand that you might send a request to either get or modify a resource and get this response back indicating that the resource cannot be returned/modified because it is locked. Is that correct?

Further, that you're saying that some other information should be provided to indicate what is required to unlock the resource. Is that right?

I think you need an example because I don't understand what the response message might look like from this.

**Review by hamishwillee — CHANGES_REQUESTED** [review]
@teoli2003 I've added some comments. I'm not sure of the right level for these responses. Generally though when I read the existing documents I can determine when I would expect the response and what it means - and some of these I'm not sure. But I haven't read this part of the spec in detail so it needs some "judgement call" from you or @Elchi3 

PS SOrry I didn't come up with this earlier - earlier I was just agreeing that we want this kind of doc.

File: files/en-us/web/http/status/102/index.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+tags:
+  - HTTP
+  - Informational
+  - Status code
+spec-urls: https://www.rfc-editor.org/rfc/rfc2518.html#section-10.1
```
Comment (hamishwillee) [inline files/en-us/web/http/status/102/index.md:8]: Thanks for fixing up all the spec sections!

File: files/en-us/web/http/status/102/index.md (hunk 6 above)
Comment (wbamberg) [reply files/en-us/web/http/status/102/index.md:3]: -> https://github.com/mdn/content/issues/22761, but it is waiting for https://github.com/mdn/content/issues/22761#issuecomment-1384085784 to be resolved. I think that should not block this PR though, we can add these pages without page types now and then set page types when we update all of HTTP.

File: files/en-us/web/http/status/226/index.md (hunk 7 above)
Comment (teoli2003) [reply files/en-us/web/http/status/226/index.md]: I have revamped this article to give context.

File: files/en-us/web/http/status/423/index.md (hunk 9 above)
Comment (teoli2003) [reply files/en-us/web/http/status/423/index.md]: I have revamped this part and added an example. I didn't go into details as we don't document the whole of WebDAV: enough to understand what is going on and the purpose here, but not enough to implement a WebDAV server or client.

File: files/en-us/web/http/status/208/index.md (hunk 5 above)
Comment (teoli2003) [reply files/en-us/web/http/status/208/index.md:13]: I have revamped this article to give more context and simplify it.

File: files/en-us/web/http/status/207/index.md (hunk 3 above)
Comment (teoli2003) [reply files/en-us/web/http/status/207/index.md]: I have revamped the page. Example of return status are useless here (it was far from being exhaustive anyway)

File: files/en-us/web/http/status/421/index.md (hunk 8 above)
Comment (teoli2003) [reply files/en-us/web/http/status/421/index.md]: I revamped this one too.

**Comment by teoli2003** [comment]
@hamishwillee 

I went again through this PR. The difficulty here is that it adds status code used in very different contexts
- Some are part of the basic HTTP semantics, like `102`
- Others are part of WebDAV and its extensions.

For the latter, I've rewritten them with the following intention:
- We should make it explicit that these are for WebDAV and not for regular HTTP (and web developers unless interacting with such servers, should never bother with them.
- I link to the glossary page about WebDAV and give minimum information (like an example) without going into too many details about the protocol: we are not documenting WebDAV here, only giving enough context so that a dev will understand what's going on there. For example, I don't describe the XML structure of the body of a WebDAV, but I use it in an example several times (as it was necessary to have an idea of the body of a response).

I hope this makes these new articles more useful.

File: files/en-us/web/http/status/208/index.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+  - Reported
+spec-urls: https://www.rfc-editor.org/rfc/rfc5842.html#section-7.1
+---
+
+{{HTTPSidebar}}
+
+> **Note:** The ability to _bind_ a resource to several paths is an extension to the {{Glossary("WebDAV")}} protocol (it may be recieved by web applications accessing a WebDAV server).
+> Browsers accessing web pages will never encounter this status code.
+
+The HTTP **`208 Already Reported`** response code is used in a {{HTTPStatus("207")}} (`207 Multi-Status`) response to save space and avoid conflicts: if the same resource is requested several times (for example as part of a collection), with different paths, only the first one is reported with {{HTTPStatus("200")}} while other responses for all other bindings will report with this `208` status code, so no conflicts are created and the response stays shorter.
```
Comment (hamishwillee) [inline files/en-us/web/http/status/208/index.md:16]:
```suggestion
The HTTP **`208 Already Reported`** response code is used in a {{HTTPStatus("207")}} (`207 Multi-Status`) response to save space and avoid conflicts.
If the same resource is requested several times (for example as part of a collection), with different paths, only the first one is reported with {{HTTPStatus("200")}}. 
Responses for all other bindings will report with this `208` status code, so no conflicts are created and the response stays shorter.
```

File: files/en-us/web/http/status/226/index.md [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+  - Status code
+  - Used
+spec-urls: https://www.rfc-editor.org/rfc/rfc3229.html#section-10.4.1
+---
+
+{{HTTPSidebar}}
+
+> **Note:** Browsers don't support _delta encoding_ with HTTP. This status code is sent back by custom servers used by specific clients.
+
+In the context of delta encodings, the HTTP **`226 IM Used`** status code is set by the server to indicate that it returns a _delta_ to the {{HTTPMethod("GET")}} request it received.
```
Comment (hamishwillee) [inline files/en-us/web/http/status/226/index.md:15]:
```suggestion
In the context of delta encodings, the HTTP **`226 IM Used`** status code is set by the server to indicate that it is returning a _delta_ to the {{HTTPMethod("GET")}} request that it received.
```

File: files/en-us/web/http/status/424/index.md [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+  - HTTP
+  - HTTP Status Code
+  - Reference
+  - Client Error
+spec-urls: https://www.rfc-editor.org/rfc/rfc4918#section-11.4
+---
+
+{{HTTPSidebar}}
+
+The HTTP **`424 Failed Dependency`** client error response code indicates that the method could not be performed on the resource because the requested action depended on another action and that action failed.
```
Comment (hamishwillee) [inline files/en-us/web/http/status/424/index.md:14]:
```suggestion
The HTTP **`424 Failed Dependency`** client error response code indicates that the method could not be performed on the resource because the requested action depended on another action, and that action failed.
```

File: files/en-us/web/http/status/226/index.md [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+spec-urls: https://www.rfc-editor.org/rfc/rfc3229.html#section-10.4.1
+---
+
+{{HTTPSidebar}}
+
+> **Note:** Browsers don't support _delta encoding_ with HTTP. This status code is sent back by custom servers used by specific clients.
+
+In the context of delta encodings, the HTTP **`226 IM Used`** status code is set by the server to indicate that it is returning a _delta_ to the {{HTTPMethod("GET")}} request that it received.
+
+Delta encoding consists in sending back to {{HTTPMethod("GET")}} requests differences (called _deltas_) from a given base document. The client uses the `A-IM:` HTTP header to indicate which difference algorithm to use and the `If-None-Match:` header to hint the server about the last version it got. The server generates a delta, sending it back in an HTTP response with the `226` status code and containing the `IM:` (with the name of the algorithm used) and `Delta-Base:` (with the {{HTTPHeader("ETag")}} matching the base document associated to the delta) HTTP headers.
```
Comment (hamishwillee) [inline files/en-us/web/http/status/226/index.md:17]: @teoli2003 The English here is hard to parse. Do you mean that you send a get request and get back a response that you can reconstuct if you know a base document it refers to?

I think you might mean:

```suggestion
With delta encoding a server responds to {{HTTPMethod("GET")}} requests with differences (called _deltas_) relative to a given base document (rather than the current document). The client uses the `A-IM:` HTTP header to indicate which differencing algorithm to use and the `If-None-Match:` header to hint the server about the last version it got. The server generates a delta, sending it back in an HTTP response with the `226` status code and containing the `IM:` (with the name of the algorithm used) and `Delta-Base:` (with the {{HTTPHeader("ETag")}} matching the base document associated to the delta) HTTP headers.
```

**Comment by hamishwillee** [comment]
@teoli2003 Looks great/good strategy. I've done some minor subedits and there is one point for clarification left.

File: files/en-us/web/http/status/226/index.md (hunk 14 above)
Comment (teoli2003) [reply files/en-us/web/http/status/226/index.md:17]: Yes, exactly.

**Review by hamishwillee — APPROVED** [review]
Thanks very much @Harshil-Jani and @teoli2003 . Looks good to me!
